packages/data-broker/src/services/dhis/DhisTranslator.js

```diff
@@ -98,14 +98,13 @@ export class DhisTranslator {
 
   translateOutboundDataValues = async (api, dataValues, dataSources) => {
     // prefetch options and types for unique data element codes so that DHIS2 doesn't get overwhelmed
+    const dataSourcesByCode = keyBy(dataSources, 'code');
     const dataElementsByCode = await this.fetchOutboundDataElementsByCode(api, dataSources);
-    return dataSources.map((dataSource, i) =>
-      this.translateOutboundDataValue(
-        dataValues[i],
-        dataSource,
-        dataElementsByCode[dataSource.dataElementCode],
-      ),
-    );
+    return dataValues.map(dataValue => {
```

Comment:
Since we now use `DataSource.find` and not `DataSource.findOrDefault`, we can no longer count on a 1-1 mapping between items in `dataValues` and `dataSources`.

This change has important implications - and would break a lot of things if left unhandled. Luckily it was caught in testing!